### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by floriangeyer21**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,3 +6,5 @@ script:
 cache:
   directories:
     - $HOME/.m2
+before_install:
```
Comment (KseniiaMakarova) [inline .travis.yml:9]: let's also add checkstyle plugin to pom.xml?

File: src/main/java/com/spingboot/demo/exceptions/FileProcessingException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.spingboot.demo.exceptions;
+
+import java.io.IOException;
+
+public class FileProcessingException extends IOException {
+
+    public FileProcessingException(String message, Throwable cause) {
+        super(message, cause);
+    }
+}
```
Comment (KseniiaMakarova) [inline src/main/java/com/spingboot/demo/exceptions/FileProcessingException.java]: empty line
and let's better extend it from RuntimeException

File: src/test/java/com/spingboot/demo/service/FileReadServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class FileReadServiceTest {
+    private String CORRECT_PATH = "src/main/resources/input.txt";
```
Comment (KseniiaMakarova) [inline src/test/java/com/spingboot/demo/service/FileReadServiceTest.java]: for test resources you should better use `src/test/resources/`

File: src/test/java/com/spingboot/demo/service/FileReadServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        Assertions.assertTrue(expectedMessage.contains(actualMessage));
+    }
+
+    @Test
+    public void withNullArgument_EmptyList() throws FileProcessingException {
+        List<String > expected = new ArrayList<>();
+        List<String> actual = fileReadService.read(null);
+        Assertions.assertEquals(expected, actual);
+    }
+}
```
Comment (KseniiaMakarova) [inline src/test/java/com/spingboot/demo/service/FileReadServiceTest.java]: empty line

File: src/test/java/com/spingboot/demo/service/FileReadServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class FileReadServiceTest {
+    private String CORRECT_PATH = "src/main/resources/input.txt";
+    private String WRONG_PATH = "wrong_path";
```
Comment (KseniiaMakarova) [inline src/test/java/com/spingboot/demo/service/FileReadServiceTest.java]: there are some requirements for constant fields

File: src/test/java/com/spingboot/demo/service/FileReadServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    public FileReadServiceTest(FileReadService fileReadService) {
+        this.fileReadService = fileReadService;
+    }
+
+    @Test
+    public void correctPath_OK() throws FileProcessingException {
+        List<String> expected = new ArrayList<>();
+        expected.add("some text for test");
+        List<String> actual = fileReadService.read(CORRECT_PATH);
+        Assertions.assertEquals(expected.toString(), actual.toString());
```
Comment (KseniiaMakarova) [inline src/test/java/com/spingboot/demo/service/FileReadServiceTest.java]:
```suggestion
        Assertions.assertEquals(expected, actual);
```
will it work?

File: src/main/java/com/spingboot/demo/service/FileReadService.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FileReadService {
```
Comment (KseniiaMakarova) [inline src/main/java/com/spingboot/demo/service/FileReadService.java]: let's have some abstract interface and make this class implement it

this class is designed to read file from the file system - so we can name it `SystemFileReaderService implements FileReaderService` for example

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/spingboot/demo/exceptions/FileProcessingException.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.spingboot.demo.exceptions;
+
+import java.io.IOException;
+
+public class FileProcessingException extends IOException {
```
Comment (KseniiaMakarova) [inline src/main/java/com/spingboot/demo/exceptions/FileProcessingException.java]: extend this exception from RuntimeException

there is hardly any point in catching IOException and then throwing another IOException

File: src/test/java/com/spingboot/demo/service/SystemFileReaderServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+import com.spingboot.demo.exceptions.FileProcessingException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.spingboot.demo.service.impl.SystemFileReaderService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
```
Comment (KseniiaMakarova) [inline src/test/java/com/spingboot/demo/service/SystemFileReaderServiceTest.java]: let's organize imports

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/spingboot/demo/service/impl/SystemFileReaderService.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SystemFileReaderService implements FileReaderService {
+
+    @Override
+    public List<String> read(String path) throws FileProcessingException {
```
Comment (Sofasmile) [inline src/main/java/com/spingboot/demo/service/impl/SystemFileReaderService.java]:
```suggestion
    public List<String> read(String path) {
```

File: src/main/java/com/spingboot/demo/service/interfaces/FileReaderService.java [hunk 11]
```diff
@@ -0,0 +1,9 @@
+package com.spingboot.demo.service.interfaces;
+
+import com.spingboot.demo.exceptions.FileProcessingException;
+import java.util.List;
+
+public interface FileReaderService {
+
+    List<String> read(String path) throws FileProcessingException;
```
Comment (Sofasmile) [inline src/main/java/com/spingboot/demo/service/interfaces/FileReaderService.java]:
```suggestion
    List<String> read(String path);
```

**Review by Sofasmile — COMMENTED** [review]
travis failed

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**